packages/perpswap/src/contracts/vault.rs

```diff
@@ -0,0 +1,130 @@
+#![allow(missing_docs)]
+use cosmwasm_std::{Addr, Uint128};
+
+// Message to instantiate the contract
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+#[serde(rename_all = "snake_case")]
+pub struct InstantiateMsg {
+    // Denomination of the USDC token
+    pub usdc_denom: String,
+
+    // Address of the factory (as string, validated later)
+    pub factory_address: String,
+
+    // Address of the USDCLP contract (as string, validated later)
```

Comment:
Is this intended to be a CW20 contract? Overall the ecosystem has moved away from CW20 to tokenfactory. But more than that, I'd recommend we just do the book keeping internally to this contract.